plugins/modules/foreman_user.py

```diff
@@ -0,0 +1,431 @@
+    required: false
+  mail:
+    description:
+      - Email address of the user
+      - Required when creating a new user
+    required: false
+  description:
+    description:
+      - Description of the user
+    required: false
+  admin:
+    description:
+      - Whether or not the user is an administrator
+    required: false
+    default: false
+    type: bool
+  user_password:
+    description:
+      - Password for the user
+    required: false
+  default_location:
+    description:
+      - The location that the user uses by default
+    required: false
+  default_organization:
+    description:
+      - The organizxation that the user uses by default
+    required: false
+  auth_source:
+    description:
+      - Authentication source where the user exists
+    required: false
+    default: Internal
+  timezone:
+    description:
+      - Timezone for the user
+      - If blank it will use the browser timezone.
+    required: false
+  locale:
+    description:
+      - The language locale for the user
+    required: false
+  roles:
+    description:
+      - List of roles assigned to the user
+    required: false
+    type: list
+  locations:
+    description:
+      - List of locations assigned to the user
+    required: false
+    type: list
+  organizations:
+    description:
+      - List of organizations assigned to the user
+    required: false
+    type: list
+  state:
+    description:
+      - State of the user
+    default: present
+    choices:
+      - present
+      - absent
+'''
+
+EXAMPLES = '''
+- name: Create a user
+  foreman_user:
+    name: test
+    firstname: Test
+    lastname: Userson
+    mail: test.userson@example.com
+    description: Dr. Test Userson
+    admin: no
+    user_password: s3cret
+    default_location: Test Location
+    default_organization: Test Organization
+    auth_source: Internal
+    timezone: Stockholm
+    locale: sv_SE
+    roles:
+      - Manager
+    locations:
+      - Test Location
+    organizations:
+      - Test Organization
+    state: present
+
+- name: Update a user
+  foreman_user:
+    name: test
+    firstname: Tester
+    state: present
+
+- name: Change password
+  foreman_user:
+    name: test
+    user_password: newp@ss
+
+'''
+
+RETURN = ''' # '''
+
+from ansible.module_utils.foreman_helper import (
+    ForemanEntityApypieAnsibleModule,
+)
+
+# This is the only true source for names (and conversions thereof)
+name_map = {
+    'name': 'login',
+    'firstname': 'firstname',
+    'lastname': 'lastname',
+    'mail': 'mail',
+    'description': 'description',
+    'admin': 'admin',
+    'user_password': 'password',
+    'default_location': 'default_location_id',
+    'default_organization': 'default_organization_id',
+    'auth_source': 'auth_source_id',
+    'timezone': 'timezone',
+    'locale': 'locale',
+    'roles': 'role_ids',
+    'locations': 'location_ids',
+    'organizations': 'organization_ids',
+}
+
+# List of allowed timezones
+timezone_list = [
```

Comment:
Are these lists (until they are generated) better suited to live somewhere in module_utils?

Reply:
If we would use them in multiple modules they would be. But I'm not sure we will do that anytime soon. Do we want to stuff a lot of into the utils that is only used in a single module?

I vote for keeping stuff in modules, and then move into module_utils when it needs to be shared between two or more modules. What do you guys think?